docs/articles/azure/azs-how-create-key-vault-ps.md

```diff
@@ -0,0 +1,97 @@
+---
+title: How to create a Key Vault - PowerShell | UKCloud Ltd
+description: Provides help for creating a Key Vault on UKCloud for Microsoft Azure
+services: azure-stack
+author: Bailey Lawson
+toc_rootlink: Users
+toc_sub1: How To
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: How to create a Key Vault - PowerShell
+toc_fullpath: Users/How To/azs-how-create-key-vault-ps.md
+toc_mdlink: azs-how-create-key-vault-ps.md
+---
+
+# How to create a Key Vault in Azure Stack with PowerShell
+
+## Overview
+
+Key Vault in Azure Stack helps safeguard cryptographic keys and secrets that cloud applications and services use, by using Key Vault, you can encrypt keys and secrets.
+
+The following process will show you how to setup a vault within Azure Stacks Key Vault, store a secret in the vault and how to extract the secret.
+
+### Prerequisites
+
+Ensure your Powershell environment is setup as detailed in [Configure PowerShell Environment and Azure Stack Modules](https://docs.ukcloud.com/articles/azure/azs-how-configure-powershell-users.html) 
+
+## Creating a new Key Vault
+
+>[!Important]
+>Values to be replaced:
+>
+>Resource Group Name: <form oninput="result.value=name.value" id="resourcegroup" >
+><input  type="text" id="name" name="name"/></form>
+>
+>Vault Name: <form oninput="result.value=name.value;result2.value=name.value;result3.value=name.value" id="vaultname">
+><input  type="text" id="name" name="name"/></form>
+>
+>Secret Value: <form oninput="result.value=name.value" id="SecretValue">
+><input  type="text" id="name" name="name"/></form>
+>
+>Secret Name: <form oninput="result.value=name.value;result2.value=name.value" id="secretname">
+><input  type="text" id="name" name="name"/></form>
+
+From your PowerShell window:
+
+<pre><code class="language-PowerShell">
+# Add environment
+Add-AzureRMEnvironment -Name 'AzureStack' -ArmEndpoint 'https://management.frn00006.azure.ukcloud.com'
+
+# Login
+Login-AzureRmAccount -EnvironmentName 'AzureStack'
+
+# Select Resource Group
+$RGName = '<output form="resourcegroup" name="result">&lt;Resource Group&gt;</output>'
+
+# Create a new vault in the Resource Group above
+New-AzureRmKeyVault -VaultName '<output form="vaultname" name="result">&lt;Vault Name&gt;</output>' -ResourceGroupName $RGName -Location 'frn00006'
+</code></pre>
+
+This will create a Key Vault in the specified resource group.
+
+## Storing a secret in the vault
+
+From your PowerShell Window:
+
+<pre><code class="language-PowerShell">

```

Comment:
Use correct markdown for code samples

Reply:
This doc uses \<pre>\<code class="language-PowerShell"> instead of \```PowerShell to allow for the \<output> tags within the code. Using triple backticks renders the output tags as part of the code instead of part of the document. Using \<pre>\<code class="language-PowerShell"> renders the tags before the code which is needed for them to work. Also when DocFX renders \```PowerShell it converts it to \<pre>\<code class="language-PowerShell"> anyway, so it doesn't make any difference once the document is published. For example, see the source code of this doc: https://docs.ukcloud.com/articles/azure/azs-how-configure-powershell-users.html